src/OidcClient/Infrastructure/LogSerializer.cs

```diff
@@ -34,9 +34,11 @@ static LogSerializer()
         /// </summary>
         /// <param name="logObject">The object.</param>
         /// <returns></returns>
-        public static string Serialize(object logObject)
+        public static string Serialize<T>(T logObject)
         {
-            return Enabled ? JsonSerializer.Serialize(logObject, JsonOptions) : "Logging has been disabled";
-        }
+            return Enabled ?
+                JsonSerializer.Serialize(logObject, (JsonTypeInfo<T>)SourceGenerationContext.Default.GetTypeInfo(typeof(T))) :
+                "Logging has been disabled";
+		}
```

Comment:
This is a breaking change because the LogSerializer can now only serialize the types in the source generator context. That works fine for our internal usage of the LogSerializer, but the LogSerializer is a public type. Anyone who is using this helper to serialize anything else will get an NRE when GetTypeInfo returns null.

Options:
- Make the LogSerializer internal. 
- Don't enable trimmable
- Make this method private, but keep the original `Serialize(object logObject)` as a public overload with `[RequiresUnreferencedCode]`